packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,302 @@
+import type { Option, Struct, u128 } from '@polkadot/types'
+import type {
+  AccountId,
+  BlockNumber,
+  Extrinsic,
+  MultiSignature,
+} from '@polkadot/types/interfaces'
+import type { AnyNumber, TypeDef } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+import { KeypairType } from '@polkadot/util-crypto/types'
+import { BN, u8aToHex, u8aToU8a } from '@polkadot/util'
+import Keyring from '@polkadot/keyring'
+
+// TODO: update with string pattern types once available
+/// A KILT-chain specific address, encoded with the KILT 38 network prefix.
+export type KiltAddress = IIdentity['address']
+/// A chain-agnostic address, which can be encoded using any network prefix.
+export type SubstrateAddress = IIdentity['address']
+
+export type Address = KiltAddress | SubstrateAddress
+
+interface ConnectionRecord extends Struct {
+  did: Address
+  deposit: Deposit
+}
+
+/// Type of signatures to link accounts to DIDs.
+export type SignatureType = MultiSignature['type']
+
+/**
+ * Type of a linking payload signing function.
+ *
+ * It takes the HEX-encoded tuple (DidIdentifier, BlockNumber) and expects the HEX-encoded signature generated by the provided address in return.
+ */
+export type LinkingSignerCallback = (
+  encodedLinkingDetails: HexString,
+  address: KiltAddress
+) => Promise<HexString>
+
+/* ### QUERY ### */
+
+export async function getAccountLinkDepositInfo(
+  linkedAccount: Address
+): Promise<Deposit | null> {
+  const { api } = await BlockchainApiConnection.getConnectionOrConnect()
+  const connectedDid = await api.query.didLookup.connectedDids<
+    Option<ConnectionRecord>
+  >(linkedAccount)
+  return connectedDid.isSome ? connectedDid.unwrap().deposit : null
+}
+
+/**
+ * Return the identifier of the DID linked to the provided account, if present.
+ *
+ * @param linkedAccount The account to use for the lookup.
+ * @returns The linked DID identifier if present, or null otherwise.
+ */
+export async function getConnectedDidForAccount(
+  linkedAccount: Address
+): Promise<IDidIdentifier | null> {
+  const { api } = await BlockchainApiConnection.getConnectionOrConnect()
+  const connectedDid = await api.query.didLookup.connectedDids<
+    Option<ConnectionRecord>
+  >(linkedAccount)
+  return connectedDid.isNone ? null : connectedDid.unwrap().did.toString()
+}
+
+/**
+ * Return all the accounts linked to the provided DID identifier.
+ *
+ * @param linkedDid The DID to use for the lookup.
+ * @param networkPrefix The optional network prefix to use to encode the returned addresses. Defaults to generic Substrate addresses (no network-specific prefix).
+ * @returns The list of accounts linked to the DID. All accounts are encoded using the KILT 38 network prefix.
+ */
+export async function getConnectedAccountsForDid(
+  linkedDid: IDidIdentifier,
+  networkPrefix: number | undefined = undefined
+): Promise<Array<KiltAddress | SubstrateAddress>> {
```

Comment:
The documentation is not accurate here rn. Shouldn't we at least default to the Kilt prefix, if we allow people to choose the address encoding at all?
@ntn-x2 @tjwelde